### PR TITLE
FUSETOOLS2-143: Create UI Tests for providing of additional Camel components settings

### PR DESCRIPTION
**Pull request by pospisilf**
# Pull Request informations

## Rebase & Merge default requirements

1. Green build for main branch
2. Wait 24 hours after PR creation
3. Green job for PR
4. Approved PR

## PR labels default process

- READY_FOR_REVIEW  &rightarrow;  REVIEW_DONE  &rightarrow;  READY_FOR_MERGE

## Tests

- [ ] Are there **Unit tests**?
- [ ] Are there **Integration tests**?
- [ ] Do we need a new **UI test**?

## PR workflow progress

1. [ ] Tagged with relevant **PR labels**
2. [ ] Green **job for PR**
3. [ ] PR was created more than **24 hours ago** or **All committers approved** it
4. [ ] Green **main** branch build

### CODE REVIEW

**Comment by pospisilf** [comment]
Blocked by https://github.com/camel-tooling/camel-lsp-client-vscode/pull/1418

File: src/ui-test/tests/camel.settings.test.ts [hunk 1]
```diff
@@ -0,0 +1,341 @@
...
+
+            await initXMLFileWithJBang(driver, FILENAME);
+
+            await waitUntilTerminalHasText(driver, `-Dcamel.jbang.version=${OLDER_JBANG_VERSION}`);
+            expect(await (await activateTerminalView()).getText()).to.contain(`-Dcamel.jbang.version=${OLDER_JBANG_VERSION}`);
+        });
+    });
+});
+
+// JBang version
```
Comment (djelinek) [inline src/ui-test/tests/camel.settings.test.ts:263]: redundant comment

File: src/ui-test/tests/camel.settings.test.ts [hunk 2]
```diff
@@ -0,0 +1,341 @@
...
+            await VSBrowser.instance.openResources(path.join(RESOURCES, CAMEL_CONTEXT_XML));
+            await waitUntilEditorIsOpened(driver, CAMEL_CONTEXT_XML);
+        });
+
+        afterEach(async function () {
+            await closeEditor(CAMEL_CONTEXT_XML, false);
+        });
+
+        it('No extra component available', async function () {
+            await testComponentInXML('abcd','abcd:xyz',false,0);
```
Comment (djelinek) [inline src/ui-test/tests/camel.settings.test.ts:309]: wrong formatting inside brackets

File: src/ui-test/tests/camel.settings.test.ts [hunk 3]
```diff
@@ -251,3 +253,82 @@ describe('User preferences', function () {
         });
     });
 });
+
+describe('Additional Camel component set test ', function () {
```
Comment (djelinek) [inline src/ui-test/tests/camel.settings.test.ts:257]: `Additional Camel components` should be sufficient label

File: src/ui-test/tests/camel.settings.test.ts [hunk 4]
```diff
@@ -251,3 +253,82 @@ describe('User preferences', function () {
...
+
+        await (await new ActivityBar().getViewControl('Explorer')).openView();
+
+    });
+
+    after(function () {
+       resetUserSettings(EXTRA_COMPONENTS_ID);
+    });
+
+    describe('Extra components user preference set test', function () {
```
Comment (djelinek) [inline src/ui-test/tests/camel.settings.test.ts:290]: redundant describe

File: src/ui-test/utils/testUtils.ts [hunk 5]
```diff
@@ -332,6 +332,17 @@ export async function setCamelCatalogVersion(version: string): Promise<void> {
 	await closeEditor('Settings', true);
 }
 
+/**
+ * Set 'Extra-components' in preferences.
+ * @param components 'Extra components' as string respecting Camel Catalog syntax.
+ */
+export async function setAdditionalComponents(components: string): Promise<void> {
```
Comment (djelinek) [inline src/ui-test/utils/testUtils.ts:339]: why we need to do this way? it is not possible to do through UI?

File: src/ui-test/tests/camel.settings.test.ts (hunk 3 above)
Comment (djelinek) [inline src/ui-test/tests/camel.settings.test.ts:257]: what about to make that part of user preferences set of tests? in CI logs it looks it is separated 
image

**Review by djelinek — CHANGES_REQUESTED** [review]

File: src/ui-test/tests/camel.settings.test.ts (hunk 3 above)
Comment (pospisilf) [reply src/ui-test/tests/camel.settings.test.ts:257]: Sure, only issue of immersion of describes. Thx for noticing!

File: src/ui-test/utils/testUtils.ts (hunk 5 above)
Comment (pospisilf) [reply src/ui-test/utils/testUtils.ts:339]: no, VS Code additional components are not added through UI :-)

File: src/ui-test/utils/testUtils.ts [hunk 6]
```diff
@@ -332,6 +332,17 @@ export async function setCamelCatalogVersion(version: string): Promise<void> {
...
+	resetUserSettings(EXTRA_COMPONENTS_ID); // Remove previous value.
+	const settingsPath = path.resolve(storageFolder, 'settings', 'User', 'settings.json');
+	const newSettings = fs.readFileSync(settingsPath, 'utf-8').slice(0, -2).concat(',', components).concat('}') // Remove '}', add ',' then additional components and give back '}'.
```
Comment (djelinek) [inline src/ui-test/utils/testUtils.ts:342]: missing `;`

**Review by djelinek — CHANGES_REQUESTED** [review]

**Review by djelinek — APPROVED** [review]

**Pull request merged**